lab#1/task#2/scripts/db.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+dbfile=../data/users.db
+inverse=$2
+
+function createFile () {
+	if [ ! -f $dbfile ]; then
+		touch $dbfile;
+	fi
+}
+
+function listItems () {
+	if [ -z $inverse ]; then
+		nl $dbfile
+	else
+		nl $dbfile | sort -nr
+	fi
+	
+}
+
+function add () {
+	createFile;
+	read -p "Enter user name: " username
+	while [ -z $username ]; # Q: How I can pass string with space without double quotes?
+	do
+		read -r -p "Please enter value: " username
+	done 
+	until [[ $username =~ [a-zA-Z] ]];
+	do
+		read -r -p "Please use latin letter only: " username
+	done
+	read -p "Enter user role: " userrole
+	while [ -z $userrole ];
+	do
+		read -r -p "Please enter value: " userrole
+	done 
+	until [[ $userrole =~ [a-zA-Z] ]];
+	do
+		read -r -p "Please use latin letters only: " userrole
+	done
+	echo "$username $userrole" >> ../data/users.db
+}
+
+function backup () {
+	date=`date +"%d-%m-%y"`
+	cp ../data/users.db ../data/${date}-users.db.backup
+}
+
+function restore () {
+	latest=$(find ../data -name '*.backup' -type f) # this not save in variable and not latest
```

Comment:
```
last_file=$(ls | egrep '.db.backup' | tail -1)
echo $last_file
```

your files will be automatically sorted by date. so last file created will be the file you need to restore.